crypto/x509v3/v3_utl.c

```diff
@@ -424,19 +424,20 @@ static STACK_OF(OPENSSL_STRING) *get_email(X509_NAME *name,
 {
     STACK_OF(OPENSSL_STRING) *ret = NULL;
     X509_NAME_ENTRY *ne;
-    ASN1_IA5STRING *email;
+    const ASN1_IA5STRING *email;
     GENERAL_NAME *gen;
     int i;
+
     /* Now add any email address(es) to STACK */
-    i = -1;
+    i = X509_NAME_get_index_by_NID(name, NID_pkcs9_emailAddress, -1);
     /* First supplied X509_NAME */
-    while ((i = X509_NAME_get_index_by_NID(name,
-                                           NID_pkcs9_emailAddress, i)) >= 0) {
+    if (i >= 0) do {
         ne = X509_NAME_get_entry(name, i);
         email = X509_NAME_ENTRY_get_data(ne);
         if (!append_ia5(&ret, email))
             return NULL;
-    }
+        i = X509_NAME_get_index_by_NID(name, NID_pkcs9_emailAddress, i);
+    } while (i >= 0);
```

Comment:
Not sure I like this. Generally speaking I am against an assignment inside an expression like in the original code here...but this is on of those occasions where it could make sense (to avoid a repetition of essentially the same line). Or if you are going to change it, the `if...do` all on one line style doesn't look right.